aten/src/ATen/core/dispatch/OperatorEntry.cpp

```diff
@@ -299,8 +299,11 @@ std::pair<const AnnotatedKernel&, const char*> OperatorEntry::computeDispatchTab
 // or alias keys and their associated keysets).
 // This function should be considered a private helper for updateDispatchTable_()
 void OperatorEntry::updateDispatchTableEntry_(const c10::Dispatcher& dispatcher, DispatchKey dispatch_key) {
-  auto dispatch_ix = static_cast<uint8_t>(dispatch_key);
-  dispatchTable_[dispatch_ix] = computeDispatchTableEntry(dispatcher, dispatch_key);
+  auto dispatch_ix = c10::getMappedDispatchKey(dispatch_key);
+  if (dispatch_ix == -1) {
```

Comment:
I'd like this condition to inline away on server and I don't see any indication that it will

Reply:
clang seems to get this right. as of C++17, it appears to be UB to get an out-of-range value into an enum, so the compiler can prove that the return value of getMappedDispatchKey is never -1. Also, the change in C++17 was a defect in the standard, so you get the correct behavior in C++14 mode as well if hte compiler implements the fix. See https://godbolt.org/z/6f194f93M and https://stackoverflow.com/questions/18195312/what-happens-if-you-static-cast-invalid-value-to-enum-class